From ee5ada90d8e66ddf3150a2e1a0ea1afe0005eb0d Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 28 Aug 2024 16:34:55 +0530
Subject: [PATCH] Add sepolicies for software gatekeeper hal

Tests done:
- Android boot in GVT-d config
- adb reboot
- Wi-Fi on/off/scan/connect/disconnect in static config

Tracked-On: OAM-124063
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 tee/disabled/hal_gatekeeper_default.te | 2 ++
 tee/disabled/service_contexts          | 1 +
 2 files changed, 3 insertions(+)
 create mode 100644 tee/disabled/hal_gatekeeper_default.te
 create mode 100644 tee/disabled/service_contexts

diff --git a/tee/disabled/hal_gatekeeper_default.te b/tee/disabled/hal_gatekeeper_default.te
new file mode 100644
index 0000000..0181578
--- /dev/null
+++ b/tee/disabled/hal_gatekeeper_default.te
@@ -0,0 +1,2 @@
+hal_attribute_service(hal_gatekeeper, hal_gatekeeper_service)
+allow hal_gatekeeper_default hal_gatekeeper_service:service_manager add;
diff --git a/tee/disabled/service_contexts b/tee/disabled/service_contexts
new file mode 100644
index 0000000..434e1fc
--- /dev/null
+++ b/tee/disabled/service_contexts
@@ -0,0 +1 @@
+android.hardware.security.sharedsecret.ISharedSecret/gatekeeper                      u:object_r:hal_gatekeeper_service:s0
-- 
2.45.2

